src/components/Docs/Docs.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect } from 'react';
+import {
+  FullType,
+  InputValue,
+  Field,
+  TypeRef,
+  DocsFirstPage,
+} from './types/types';
+
+import { DocsPage } from './DocsPage/DocsPage';
+import { getIntrospectionQueryData } from '../../utils/APIutils';
+import { useTranslation } from 'react-i18next';
+
+import { Button, Container, Title } from '@mantine/core';
+
+export function Docs() {
+  const { t } = useTranslation();
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isShown, setIsShown] = useState(false);
+  const [schemaTypes, setSchemaTypes] = useState<null | FullType[]>(null);
+  const [currentShownElement, setCurrentShownElement] = useState<
+    null | FullType | InputValue | Field | TypeRef | DocsFirstPage
+  >(null);
+  const [history, setHistory] = useState<(FullType | Field | DocsFirstPage)[]>(
+    []
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await getIntrospectionQueryData();
+      const schema_types = (data.__schema.types as FullType[]).filter(
+        (type) => !type.name.startsWith('__')
+      );
+      setSchemaTypes(schema_types);
+      const startElem: DocsFirstPage = {
+        name: 'Docs',
+        description:
+          'A GraphQL schema provides a root type for each kind of operation.',
+        rootType: schema_types[0],
+      };
+      setCurrentShownElement(startElem);
+      setHistory([startElem]);
+    };
+    fetchData()
+      .then(() => {
+        setIsLoaded(true);
+      })
+      .catch(console.error);
```

Comment:
setIsLoaded can be called from fetchData itself (same as setHistory, setCurrentShownElement)